java/component/keyple-plugin/remotese/src/main/java/org/eclipse/keyple/plugin/remotese/pluginse/VirtualObservableReaderImpl.java

```diff
@@ -36,6 +47,75 @@ public VirtualObservableReaderImpl(VirtualReaderSession session, String nativeRe
     }
 
 
+    @Override
+    public void addObserver(ObservableReader.ReaderObserver observer) {
+        if (observer == null) {
+            return;
+        }
+
+        logger.trace("Adding '{}' as an observer of '{}'.", observer.getClass().getSimpleName(),
+                getName());
+
+        synchronized (sync) {
+            if (observers == null) {
+                observers = new HashSet<ReaderObserver>(1);
+            }
+            observers.add(observer);
+        }
+    }
+
+    @Override
+    public void removeObserver(ObservableReader.ReaderObserver observer) {
```

Comment:
in removeObserver(ObservableReader.ReaderReaderObserver observer), the parameter isn't defined as final